### PR TITLE
add CLI argument `--disable-workspace-trust`

### LINKED ISSUE #5288: [Feat]: Way to invoke code-server with --disable-workspace-trust option.
## What is your suggestion?

VS Code now has the workspace trust feature which requires a workspace to be trusted before working in it. Upstream VS Code does have a command line option `--disable-workspace-trust` to disable this feature:

* https://code.visualstudio.com/docs/editor/workspace-trust#_commandline-switch

but the same option is not available in `code-server`. Further, adding settings to:

* ~/.local/share/code-server/User/settings.json

such as:

```
    "security.workspace.trust.banner": "never",
    "security.workspace.trust.untrustedFiles": "open"
```

have no effect since VS Code settings in the main are stored in the browser now and file based settings aren't used.

The ask is for a way to be able to disable workspace trust mechanism from the command line when running code-server.

## Why do you want this feature?

Am embedding code-server in another hosted web application where precisely controlling what files are available and users needing to trust the workspace is not a great workflow and is unnecessary.

Further, we are using a VS Code extension which allows remote API access to automate actions in VS Code from the web application code-server is embedded in, such as open a file, but this extension's workflow is broken by the workspace trust feature because a file cannot be opened via the remote API until the workspace is trusted.

## Are there any workarounds to get this functionality today?

No idea.

## Are you interested in submitting a PR for this?

I have no knowledge on how it would be added.

### PR DESCRIPTION
**Pull request by kotborealis**
Add CLI flag to disable workspace trust feature, as described in documentation:
https://code.visualstudio.com/docs/editor/workspace-trust

Fixes #5288

### CODE REVIEW

**Comment by kotborealis** [comment]
Sorry for the mess with force-pushes, could not get commit signature right 😢

**Comment by jsjoeio** [comment]
Thank you! Do you mind adding or updating a test?

**Comment by jsjoeio** [comment]
And if it's not too much trouble, can you post a video showing this working?

**Comment by kotborealis** [comment]
Okay, I'll find some time today for tests and video 👍🏻

**Comment by kotborealis** [comment]
[Demo (youtube)](https://www.youtube.com/watch?v=16dg6kJlBBg)
Sorry for screen tearing, drivers won't make friends with screen recorder.

**Comment by jsjoeio** [comment]
That video was awesome!! Thanks for doing it. Looks great 🚀

File: test/e2e/trust.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { clean } from "../utils/helpers"
```
Comment (jsjoeio) [inline test/e2e/trust.test.ts:1]: Looks like we can remove this!

File: test/e2e/trust.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  test("should see the 'I Trust...' option", async ({ codeServerPage }) => {
+    expect(await codeServerPage.page.isVisible("text=Yes, I trust")).toBe(true)
+  })
+})
+
+describe("Workspace trust (disabled)", ["--disable-workspace-trust"], {}, async () => {
+  test("should not see the 'I Trust...' option", async ({ codeServerPage }) => {
+    expect(await codeServerPage.page.isVisible("text=Yes, I trust")).toBe(false)
+  })
+})
```
Comment (jsjoeio) [inline test/e2e/trust.test.ts:14]: Thanks for adding these tests!

**Review by jsjoeio — APPROVED** [review]
Great work!

File: test/e2e/trust.test.ts (hunk 1 above)
Comment (kotborealis) [reply test/e2e/trust.test.ts]: Done

**Pull request merged**